### PR TITLE
Upgrade `ignore` dependency to version 5

### LINKED ISSUE #92: Update `ignore` package to latest version
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/globby/issues/92)

> There is a $20.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/globby/issues/92).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/globby/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by yaodingyd**
Fix #92

### CODE REVIEW

File: gitignore.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const reduceIgnore = files => {
 	}, gitIgnore());
 };
 
-const getIsIgnoredPredecate = (ignores, cwd) => {
+const getIsIgnoredPredicate = (ignores, cwd) => {
```
Comment (sindresorhus) [inline gitignore.js]: Why change this to something that is incorrect?

**Comment by sindresorhus** [comment]
I haven't look into this closely, but I always get suspicious when you have to modify tests to get something working. Are you sure this is the correct way to solve it? I would have thought we needed to do some changes to `gitignore.js`, like in https://github.com/sindresorhus/globby/blob/71b9c58f86b4625b58ae241360016f4418dac288/gitignore.js#L48

**Comment by sindresorhus** [comment]
https://github.com/kaelzhang/node-ignore#upgrade-4x---5x

**Comment by yaodingyd** [comment]
The 5x breaking change is that for every path it uses `const REGEX_TEST_INVALID_PATH = /^\.*\/|^\.+$/` this regex to check invalid path and throws errors. So in the test case for "foo.js" after `path.relative` it returns `../../foo.js` which throws errors. I want to confirm if globby is used, all the file paths from fastglob would be relative to current cwd and we would not have this issue, and all I need to do is take care of the test file? 

Or we do need to consider this kind of path so in line 48 we need to check if path matches that regex?

**Comment by sindresorhus** [comment]
> Or we do need to consider this kind of path so in line 48 we need to check if path matches that regex?

I don't know. Part of the task of fixing #92 is figuring that out.

**Comment by sindresorhus** [comment]
@yaodingyd Did you figure it out?

**Comment by yaodingyd** [comment]
I'm checking the input path: if it matches a relative path starting with `./` then I get relative path to `cwd`; otherwise I would pass it directly to `ignore`. I think my assumption makes sense: usually when we use globby, we would use it under `cwd` and any files path looks like the test ones, they should be under `cwd`, which is the results of `path.relative`, so that we can directly use them.

**Comment by sindresorhus** [comment]
Closing in favor of #20

**Pull request closed without merging**